Add authority claimer documentation to node (#143)
cartesi/rollups-node: README.md

> The [**Dispatcher**](./offchain/dispatcher/README.md) is the component that messages *Inputs* via the **Broker** to be processed elsewhere and submits *Claims* to the blockchain.

Reviewer: ```suggestion
The [**Dispatcher**](./offchain/dispatcher/README.md) is the component that messages *Inputs* via the **Broker** to be processed later on.
```

> The [**Authority Claimer**](./offchain/authority-claimer/README.md) is the component that submits *Claims* to the blockchain at the end of every epoch.

Reviewer: ```suggestion
The [**Authority Claimer**](./offchain/authority-claimer/README.md) is the component that submits *Claims* to the blockchain at the end of each epoch that contains at least one *Input*.
```

The [**Advance Runner**](./offchain/advance-runner/README.md) is the one responsible for relaying *Inputs* to the **Server Manager** to be processed by the underlying DApp running in the embedded **Cartesi Machine**.
The **Advance Runner** obtains the resulting **Outputs** and **Claims** from the **Server Manager** and adds them to the **Broker**.

The [**Indexer**](./offchain/indexer/README.md) consumes all *Inputs* and *Outputs* transferred by the **Broker** and stores them in a **PostgreSQL** database for later querying via a **GraphQL Server** instance.

The [**Inspect Server**](./offchain/inspect-server/README.md) is responsible for the processing of *Inspect* requests via HTTP.

## Framework features

> The Cartesi Node contains 3 main features, the combination of which defines the node capabilities

Reviewer: ```suggestion
The Cartesi Node contains 3 main features, the combination of which defines the node capabilities.
```

### Reader Feature

> With the Reader Feature, any *Input* read from the blockchain by the **State-fold Server** is received by the **Dispatcher** and relayed to the **Broker** through an input-specific stream.

Reviewer 2: This paragraph seems strange because the text describes details of the implementation instead of the feature itself.
We first need an explanatory paragraph for each feature. For instance, in the processing inputs feature, we could have something like:

> The Node is capable of processing rollup inputs stored in the blockchain. To do that, the node gets the inputs from the blockchain, sends them to the Cartesi machine, and stores the result in the node database. Once the input result is in the database, the node client can query the outputs in the GraphQL API.

The *Input* is eventually consumed by the **Advance Runner** and used to advance the state of the **Server Manager**, thus generating an *Advance* request to be processed by the underlying DApp.

After finishing the processing, the DApp may generate a number of *Outputs*, which are eventually retrieved by the **Advance Runner** and fed back into the **Broker** through an output-specific stream.

*Inputs* and *Outputs* are consumed from their respective streams by the **Indexer** and stored in a **PostgreSQL** database and may be queried by the DApp **Frontend** via a **GraphQL Server** instance.

### Claiming Feature

The Claiming Feature allows a node to be run as a *Validator* node.

> It complements the Reader Feature by generating *Claims* at the end of *Epochs*, which are generated by the **Server Manager**, and sent to the **Broker** by the **Advance Runner** through a claims-specific stream to be eventually consumed by the **Authority Claimer** for being submitted to the blockchain.

Reviewer: Should we stress that claims are only sent when there were inputs processed within said epochs?

Author: I added this info. I think it costs little to the reader and cover our bases so there's less confusion later.

### Inspect Data Feature

Every *Inspect* request sent by the **Frontend** via the Rollups HTTP API is captured by the **Inspect Server** and forwarded to the **Server Manager**, which queries the state of the underlying DApp via *Inspect* requests.

Every request results in an *Inspect Response* that is returned to the **Inspect Server**, which sends it back to the **Frontend**.

## Execution modes

> The Cartesi Node may act as a *User* (aka *Reader*), as a *Validator* or run locally as a *Host*, hence performing different roles, which may be represented by different workflows.

Reviewer 2: ```suggestion
The Cartesi Node may act as a *Reader*, or a *Validator*, or run locally as a *Host*, performing different roles, which different workflows may represent.
```

I removed the "User" part; I think it is confusing. I also applied some suggestions from Grammarly.

I'm still not sure what "different workflows may represent" means.

### Reader Mode

> The *Reader Mode* is the default running operation of the node. It doesn't submit claims to the blockchain, instead only reads and updates the state. It contains the Reader and Inspect Data features.

Reviewer 2: Each sentence should have its own line so the diffs later are easier to read.

> The *Reader Mode* is the default running operation of the node. It doesn't submit claims to the blockchain, instead only reads and updates the state. It contains the Reader and Inspect Data features.

Reviewer 2: > The *Reader Mode* is the default running operation of the node.

I'm afraid I have to disagree. We shouldn't provide a default mode because the user should be conscious of his actions.

Author: Fair enough, by default I meant that every node will run this logic, even a validator node that has the claiming as well. But I can see now I did not made myself clear.

> The *Reader Mode* is the default running operation of the node. It doesn't submit claims to the blockchain, instead only reads and updates the state. It contains the Reader and Inspect Data features.

Reviewer: Cosmetic change, to help with diffs:

```suggestion
The *Reader Mode* is the default running operation of the node.
It doesn't submit claims to the blockchain, instead only reads and updates the state.
It contains the Reader and Inspect Data features.
```
